wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java

```diff
@@ -656,6 +667,41 @@ public void deleteProject(@ApiParam("The workspace id")
         }
     }
 
+    @GET
+    @Path("/{id}/check")
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(value = "Get state of the workspace agent by the workspace id and agent id")
+    @ApiResponses({@ApiResponse(code = 200, message = "The response contains requested workspace entity"),
+                   @ApiResponse(code = 400, message = "Missed required parameters, parameters are not valid"),
+                   @ApiResponse(code = 403, message = "The user is not workspace owner"),
+                   @ApiResponse(code = 404, message = "The workspace with specified id does not exist"),
+                   @ApiResponse(code = 500, message = "Internal server error occurred")})
+    public WsAgentHealthStateDto checkAgentHealth(@ApiParam(value = "Workspace id")
+                                                  @PathParam("id") String key) throws NotFoundException,
+                                                                                             ServerException,
+                                                                                             ForbiddenException,
+                                                                                             BadRequestException,
```

Comment:
It should not be thrown by REST this service, so should not be exposed in this method